### ISSUE TITLE
errors updating nginx in debian bullseye to the latest version

### ISSUE THREAD

**Issue by WalterPrech** [oerdnj/deb.sury.org#2150]
**Frequently asked questions**

- [x ] I have read [Frequently Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)
- [x ] I have looked at the list of the existing issues (including closed issues) and searched if my issue has been already reported
- [ ] I have tried to resolve the issue myself and will describe what I did in clear and concise manner

**Describe the bug**
Post-installation script of nginx throws the following errors:
```
dpkg: error processing package nginx (--configure):
 installed nginx package post-installation script subprocess returned error exit status 1
dpkg: dependency problems prevent configuration of libnginx-mod-http-image-filter:
 libnginx-mod-http-image-filter depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-image-filter (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-stream:
 libnginx-mod-stream depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-stream (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-upstream-fair:
 libnginx-mod-http-upstream-fair depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-upstream-fair (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-ssl-ct:
 libnginx-mod-ssl-ct depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-ssl-ct (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of nginx-common:
 nginx-common depends on nginx (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package nginx is not configured yet.
 nginx-common depends on nginx (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package nginx is not configured yet.

dpkg: error processing package nginx-common (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-stream-geoip:
 libnginx-mod-stream-geoip depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.
 libnginx-mod-stream-geoip depends on libnginx-mod-stream (= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package libnginx-mod-stream is not configured yet.

dpkg: error processing package libnginx-mod-stream-geoip (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of nginx-full:
 nginx-full depends on libnginx-mod-http-upstream-fair (>= 1:0.0~git20120408.a18b409+1.26.1~); however:
  Package libnginx-mod-http-upstream-fair is not configured yet.
 nginx-full depends on nginx (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package nginx is not configured yet.
 nginx-full depends on nginx (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package nginx is not configured yet.

dpkg: error processing package nginx-full (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-xslt-filter:
 libnginx-mod-http-xslt-filter depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-xslt-filter (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-auth-pam:
 libnginx-mod-http-auth-pam depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-auth-pam (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-geoip:
 libnginx-mod-http-geoip depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-geoip (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-brotli-filter:
 libnginx-mod-http-brotli-filter depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-brotli-filter (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-echo:
 libnginx-mod-http-echo depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-echo (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-subs-filter:
 libnginx-mod-http-subs-filter depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-subs-filter (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-geoip2:
 libnginx-mod-http-geoip2 depends on libnginx-mod-stream (<< 1.26.1.1~); however:
  Package libnginx-mod-stream is not configured yet.
 libnginx-mod-http-geoip2 depends on libnginx-mod-stream (>= 1.26.1); however:
  Package libnginx-mod-stream is not configured yet.
 libnginx-mod-http-geoip2 depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-geoip2 (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-stream-geoip2:
 libnginx-mod-stream-geoip2 depends on libnginx-mod-stream (<< 1.26.1.1~); however:
  Package libnginx-mod-stream is not configured yet.
 libnginx-mod-stream-geoip2 depends on libnginx-mod-stream (>= 1.26.1); however:
  Package libnginx-mod-stream is not configured yet.
 libnginx-mod-stream-geoip2 depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-stream-geoip2 (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of nginx-core:
 nginx-core depends on libnginx-mod-http-geoip (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package libnginx-mod-http-geoip is not configured yet.
 nginx-core depends on libnginx-mod-http-geoip (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package libnginx-mod-http-geoip is not configured yet.
 nginx-core depends on libnginx-mod-http-image-filter (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package libnginx-mod-http-image-filter is not configured yet.
 nginx-core depends on libnginx-mod-http-image-filter (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package libnginx-mod-http-image-filter is not configured yet.
 nginx-core depends on libnginx-mod-http-xslt-filter (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package libnginx-mod-http-xslt-filter is not configured yet.
 nginx-core depends on libnginx-mod-http-xslt-filter (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package libnginx-mod-http-xslt-filter is not configured yet.
 nginx-core depends on libnginx-mod-stream (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package libnginx-mod-stream is not configured yet.
 nginx-core depends on libnginx-mod-stream (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package libnginx-mod-stream is not configured yet.
 nginx-core depends on libnginx-mod-stream-geoip (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package libnginx-mod-stream-geoip is not configured yet.
 nginx-core depends on libnginx-mod-stream-geoip (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package libnginx-mod-stream-geoip is not configured yet.
 nginx-core depends on nginx (>= 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe); however:
  Package nginx is not configured yet.
 nginx-core depends on nginx (<< 1.26.1-1+0~20240604.35+debian11~1.gbpca5abe.1~); however:
  Package nginx is not configured yet.

dpkg: error processing package nginx-core (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-dav-ext:
 libnginx-mod-http-dav-ext depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-dav-ext (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-mail:
 libnginx-mod-mail depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-mail (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libnginx-mod-http-brotli-static:
 libnginx-mod-http-brotli-static depends on nginx-abi-1.26.1-1; however:
  Package nginx-abi-1.26.1-1 is not installed.
  Package nginx which provides nginx-abi-1.26.1-1 is not configured yet.

dpkg: error processing package libnginx-mod-http-brotli-static (--configure):
 dependency problems - leaving unconfigured
Processing triggers for libc-bin (2.31-13+deb11u10) ...
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for php8.2-cli (8.2.20-2+0~20240608.57+debian11~1.gbp6b5ea6) ...
Processing triggers for php8.3-cli (8.3.8-2+0~20240608.35+debian11~1.gbp4d7dbc) ...
Processing triggers for php8.2-fpm (8.2.20-2+0~20240608.57+debian11~1.gbp6b5ea6) ...
Errors were encountered while processing:
 nginx
 libnginx-mod-http-image-filter
 libnginx-mod-stream
 libnginx-mod-http-upstream-fair
 libnginx-mod-ssl-ct
 nginx-common
 libnginx-mod-stream-geoip
 nginx-full
 libnginx-mod-http-xslt-filter
 libnginx-mod-http-auth-pam
 libnginx-mod-http-geoip
 libnginx-mod-http-brotli-filter
 libnginx-mod-http-echo
 libnginx-mod-http-subs-filter
 libnginx-mod-http-geoip2
 libnginx-mod-stream-geoip2
 nginx-core
 libnginx-mod-http-dav-ext
 libnginx-mod-mail
 libnginx-mod-http-brotli-static
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**To Reproduce**
```
sudo apt update
sudo apt upgrade
```

**Your understanding of what is happening**
The post-installation script in not working well.

**Expected behavior**
Installation of nginx and php with all dependencies should work without an error

**Distribution (please complete the following information):**
 - OS: Debian bullseye
 - Architecture: amd64
 - Repository: 
 nginx.list: https://packages.sury.org/nginx/ bullseye main
 php.list: https://packages.sury.org/php/ bullseye main

**Package(s) (please complete the following information):**
```
Package files:
 100 /var/lib/dpkg/status
     release a=now
 500 https://packages.sury.org/php bullseye/main amd64 Packages
     release o=deb.sury.org,a=bullseye,n=bullseye,c=main,b=amd64
     origin packages.sury.org
 500 https://packages.sury.org/nginx bullseye/main amd64 Packages
     release o=deb.sury.org,a=bullseye,n=bullseye,c=main,b=amd64
     origin packages.sury.org
 500 http://downloads.mariadb.com/Tools/debian bullseye/main amd64 Packages
     release o=MariaDB Enterprise Tools,n=bullseye,l=MariaDB,c=main,b=amd64
     origin downloads.mariadb.com
1000 https://dlm.mariadb.com/repo/maxscale/latest/apt bullseye/main arm64 Packages
     release o=MariaDB Enterprise,n=bullseye,l=MariaDB Enterprise,c=main,b=arm64
     origin dlm.mariadb.com
1000 https://dlm.mariadb.com/repo/maxscale/latest/apt bullseye/main amd64 Packages
     release o=MariaDB Enterprise,n=bullseye,l=MariaDB Enterprise,c=main,b=amd64
     origin dlm.mariadb.com
1000 https://dlm.mariadb.com/repo/mariadb-server/10.11/repo/debian bullseye/main arm64 Packages
     release o=MariaDB,n=bullseye,l=MariaDB,c=main,b=arm64
     origin dlm.mariadb.com
1000 https://dlm.mariadb.com/repo/mariadb-server/10.11/repo/debian bullseye/main amd64 Packages
     release o=MariaDB,n=bullseye,l=MariaDB,c=main,b=amd64
     origin dlm.mariadb.com
 500 https://cli.github.com/packages stable/main amd64 Packages
     release o=gh,n=stable,l=gh,c=main,b=amd64
     origin cli.github.com
 500 http://security.debian.org bullseye-security/main amd64 Packages
     release v=11,o=Debian,a=oldstable-security,n=bullseye-security,l=Debian-Security,c=main,b=amd64
     origin security.debian.org
 500 http://ftp.debian.org/debian bullseye-updates/main amd64 Packages
     release v=11-updates,o=Debian,a=oldstable-updates,n=bullseye-updates,l=Debian,c=main,b=amd64
     origin ftp.debian.org
 500 http://ftp.debian.org/debian bullseye/contrib amd64 Packages
     release v=11.9,o=Debian,a=oldstable,n=bullseye,l=Debian,c=contrib,b=amd64
     origin ftp.debian.org
 500 http://ftp.debian.org/debian bullseye/main amd64 Packages
     release v=11.9,o=Debian,a=oldstable,n=bullseye,l=Debian,c=main,b=amd64
     origin ftp.debian.org
Pinned packages:
```

**Comment by oerdnj** [comment]
This is not a full log; it's impossible to tell what went wrong in this case.

**Comment by endoze** [comment]
@WalterPrech I ran into this issue myself on an Ubuntu 20.04 machine. In order to get everything working again on the box, I used `sudo dpkg --purge --force-all --no-triggers nginx` first to remove nginx. After that, I removed all of the broken/unmet dependency packages with `sudo apt remove -y <all of the packages from the error message here>`.

After I removed everything, I did an update/upgrade to ensure everything else on the box was up to date. Then I re-installed nginx via `sudo apt install -y nginx`. I can only guess as to the cause, but I'd imagine some sort of post-install script ran into an issue and error'd out. This left the nginx package unconfigured which prevented everything from working properly. A full remove and re-install sorted it out for me. So something about upgrading an existing install is maybe broken in the post-install script.
